aws/data_source_aws_transfer_server_test.go

```diff
@@ -0,0 +1,272 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataSourceAwsTransferServer_basic(t *testing.T) {
+	resourceName := "aws_transfer_server.test"
+	datasourceName := "data.aws_transfer_server.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsTransferServerConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceAwsTransferServerCheck(datasourceName, resourceName),
```

Comment:
We can remove the custom function by using [`resource.TestCheckResourceAttrPair()`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrPair), e.g.

```suggestion
					resource.TestCheckResourceAttrPair(datasourceName, "arn", resourceName, "arn"),
					resource.TestCheckResourceAttrPair(datasourceName, "endpoint", resourceName, "endpoint"),
					resource.TestCheckResourceAttrPair(datasourceName, "identity_provider_type", resourceName, "identity_provider_type"),
					resource.TestCheckResourceAttrPair(datasourceName, "invocation_role", resourceName, "invocation_role"),
					resource.TestCheckResourceAttrPair(datasourceName, "logging_role", resourceName, "logging_role"),
					resource.TestCheckResourceAttrPair(datasourceName, "url", resourceName, "url"),
```

Along with simplifying the code, we prefer using this upstream function since it also has some special handling in the upcoming Terraform 0.12 Provider SDK to ignore differences between missing values and zero-values. 😄